addons/mrp_subcontracting/wizard/stock_picking_return.py

```diff
@@ -19,7 +19,7 @@ def _compute_subcontract_location_id(self):
     @api.onchange('picking_id')
     def _onchange_picking_id(self):
         res = super(ReturnPicking, self)._onchange_picking_id()
-        if any(self.product_return_moves.filtered(lambda r: r.quantity > 0).move_id.mapped('is_subcontract')):
+        if any(wizard_line.quantity > 0 and wizard_line.move_id.is_subcontract for wizard_line in self.product_return_moves):
```

Comment:
detail, `wizard_line` is not very clear (technical), prefer something explaining the meaning of the record, e.g. `return_line`